internal/observability/trace/trace.go

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package trace provides trace functions.
+package trace
+
+import (
+	"context"
+
+	"go.opencensus.io/trace"
+)
+
+var (
+	enabled bool
+)
+
+type Tracer interface {
+	Start(ctx context.Context)
+}
+
+type tracer struct {
+	samplingRate float64
+}
+
+func StartSpan(ctx context.Context, name string, opts ...trace.StartOption) (context.Context, *trace.Span, func()) {
+	if !enabled {
+		return ctx, nil, func() {}
+	}
+
+	ctx, span := trace.StartSpan(ctx, name, opts...)
+	end := func() {
+		span.End()
+	}
+	return ctx, span, end
```

Comment:
why did you divide span and end func?

Reply:
if we didn't divide `span` and `end`, we should check whether the `span` is nil or not in the caller's defer function.

Reply:
but you didn't check span is nil or not in trace.go#L45

Reply:
my opinion is below
```go
func StartSpan(ctx context.Context, name string, opts ...trace.StartOption) (context.Context, *trace.Span, err) {
	if !enabled {
		return ctx, errors.ErrTracerDisabled
	}

	ctx, span := trace.StartSpan(ctx, name, opts...)
	if span == nil {
		return ctx, nil, errors.ErrTracerSpanNotFound
	}
	return ctx, span, nil
}
```